packages/client/lib/rpc/modules/engine.ts

```diff
@@ -277,17 +349,21 @@ export class Engine {
    *   3. validationError: String|null - validation error message
    */
   async newPayloadV1(params: [ExecutionPayloadV1]): Promise<PayloadStatusV1> {
-    const [payloadData] = params
-    const {
-      blockNumber: number,
-      receiptsRoot: receiptTrie,
-      prevRandao: mixHash,
-      feeRecipient: coinbase,
-      transactions,
-      parentHash,
-      blockHash,
-    } = payloadData
-    const { chainCommon: common } = this.config
+    const [payload] = params
+    const { parentHash, blockHash } = payload
+
+    const { block, error } = await assembleBlock(payload, this.chain)
+    if (!block || error) {
+      let response = error
+      if (!response) {
```

Comment:
If I understand this correctly, the function `assembleBlock` will return **or** a block **or** an error - Probably I am wrong but I feel that L359-L362 is unreachable?

I feel that I am most likely wrong here tho

Reply:
"will return or a block or an error"

Is there a word or something missing between "return" and "or"? 🙂

Reply:
@holgerd77 negative! what I meant is that it wont return both, but only one of them - does that clarify things?

Reply:
Ah, that would be "either a block or an error".

Yes, thanks, that clarifies. 😄

Reply:
I am returning an object with optional properties so I needed this additional !block check otherwise it thinks it might be undefined.

Let me know if there is another way to do it, the OR case was my previous implementation but you said it was confusing to assign the error result to "block"